### PR TITLE
Order and order product

### PR DESCRIPTION
**Pull request by thientrankhh**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ yarn-error.log
 .idea
 node_modules
 key.json
+TestController.php
```
Comment (DungLe1603) [inline .gitignore]: remove controller from .gitignore.
if TestController isn't used, remove it from your source code

File: app/Http/Controllers/Admin/OrderController.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use App\Model\Order;
+use App\Model\OrderProduct;
+use App\Model\Product;
+
+class OrderController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('checkLoginAdmin');
+    }
```
Comment (DungLe1603) [inline app/Http/Controllers/Admin/OrderController.php]: middleware should be place in route.
Ex:
    Route::group(['prefix' => 'admin', 'namespace' => 'Admin', middleware => 'loginAdmin' ], function () {
        Route::get(...);
    }

File: app/Http/Controllers/Admin/ProductController.php [hunk 3]
```diff
@@ -21,32 +21,36 @@ public function __construct()
...
             'description' => $request->description,
             'configuration' => $request->configuration,
             'price' => $request->price,
-            'images' => $image,
-            'created_at' => Carbon::now('Asia/Ho_Chi_Minh'),
-        ]);
-        $newProduct->save();
+            'images' => $image
+        ];
+        Product::create($data);
```
Comment (DungLe1603) [inline app/Http/Controllers/Admin/ProductController.php:57]: use fillable, not necessary to declare $data = [...]
EX:
 Product::create($request->all());

File: app/Http/Controllers/Admin/OrderController.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    public function showOrder()
+    {
+        $orders = Order::paginate(8);
+
+        return view('admin.order.listOrder', compact('orders'));
+    }
```
Comment (DungLe1603) [inline app/Http/Controllers/Admin/OrderController.php:18]: show() is often used to display a specific order.
with list order, you should you list or index.

File: app/Http/Requests/CreateProductRequest.php [hunk 5]
```diff
@@ -25,23 +25,11 @@ public function rules()
...
-    {
-        return [
-            'product_name.required' => 'Product Name Not Null',
-            'quantity.required' => 'Quantity Not Null',
-            'description.required' => 'Description Not Null',
-            'configuration.required' => 'Configuration Not Null',
-            'price.required' => 'Price Not Null',
-            'image.required' => "You don't have file",
+            'price' => 'required|numeric|min:1|max:1000',
+            'image' => 'required|image:jpeg,jpg,png'
```
Comment (DungLe1603) [inline app/Http/Requests/CreateProductRequest.php:32]: price with max=1000? 
There is the headphone costing 100 millions vnd

File: resources/views/admin/dashboard.blade.php [hunk 6]
```diff
@@ -97,4 +32,4 @@
         <!-- /.content -->
     </div>
     <!-- /.content-wrapper -->
-@endsection
\ No newline at end of file
+@endsection
```
Comment (DungLe1603) [inline resources/views/admin/dashboard.blade.php]: Add a blank line

File: resources/views/admin/layouts/footer.blade.php [hunk 7]
```diff
@@ -1,13 +1,9 @@
...
+
 </footer>
 
 <!-- Control Sidebar -->
 <aside class="control-sidebar control-sidebar-dark">
     <!-- Control sidebar content goes here -->
 </aside>
-<!-- /.control-sidebar -->
\ No newline at end of file
+<!-- /.control-sidebar -->
```
Comment (DungLe1603) [inline resources/views/admin/layouts/footer.blade.php]: add a blank line

File: routes/web.php [hunk 8]
```diff
@@ -44,11 +44,15 @@
...
-    Route::get('/product', 'ProductController@listAllProducts')->name('listAllProducts');
+    Route::get('/product', 'ProductController@index')->name('index');
+    Route::get('/create', 'ProductController@create')->name('create');
     Route::post('/store', 'ProductController@store')->name('store');
     Route::get('/edit/{product}', 'ProductController@editProduct')->name('editProduct');
     Route::put('/update/{id}', 'ProductController@update')->name('update');
     Route::get('/exportProduct', 'ProductController@exportProduct')->name('exportProduct');
     Route::post('/importProduct', 'ProductController@importProduct')->name('importProduct');
-    Route::delete('/destroyProduct/{id}', 'ProductController@destroyProduct')->name('destroyProduct');
+    Route::delete('/delete/{id}', 'ProductController@destroy')->name('destroy');
```
Comment (DungLe1603) [inline routes/web.php]: Should name clearly
EX:
    '/update/{id}' => '/product/{id}/update'
    '/delete/{id}' => '/product/{id}/delete'
But I suggest using resource:
  Route::resource('products', 'ProductController')->middleware('checkLoginAdmin');

File: app/Http/Requests/CreateProductRequest.php (hunk 5 above)
Comment (thientrankhh) [reply app/Http/Requests/CreateProductRequest.php:32]: Produced by Thien , High Quality :D :v
Để anh fix lại :)))))))))

File: app/Http/Controllers/Admin/OrderController.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    public function orderProduct($id)
+    {
+        $orderProduct = OrderProduct::with('order')->findOrFail($id)->get();
+        $products = Product::all();
+        $data = [
```
Comment (szabogaliakos) [inline app/Http/Controllers/Admin/OrderController.php]: Consider the usages of compact('data')

File: app/Http/Controllers/Admin/ProductController.php [hunk 10]
```diff
@@ -16,50 +16,81 @@
...
-    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
 
-    public function index()
+    public function index(Product $products)
     {
```
Comment (DungLe1603) [inline app/Http/Controllers/Admin/ProductController.php:26]: Declare $product variable but not used?

File: app/Http/Requests/CreateProductRequest.php [hunk 11]
```diff
@@ -24,11 +24,11 @@ public function authorize()
...
     {
         return [
-            'product_name' => 'required',
-            'quantity' => 'required|numeric|min:1|max:100',
+            'product_name' => 'required|unique:products',
+            'quantity' => 'required|numeric|min:0|max:100',
             'description' => 'required',
             'configuration' => 'required',
-            'price' => 'required|numeric|min:1|max:1000',
+            'price' => 'required|numeric|min:1|max:300',
```
Comment (DungLe1603) [inline app/Http/Requests/CreateProductRequest.php]: 300?

File: app/Http/Requests/UpdateProductRequest.php [hunk 12]
```diff
@@ -24,11 +25,12 @@ public function authorize()
...
         return [
+//            'product_name' => ['required', Rule::unique('products')->ignore('id')],
             'product_name' => 'required',
-            'quantity' => 'required|numeric|min:1|max:100',
+            'quantity' => 'required|numeric|min:0|max:100',
             'description' => 'required',
             'configuration' => 'required',
-            'price' => 'required|numeric|min:1|max:1000',
+            'price' => 'required|numeric|min:1|max:300',
             'image' => 'image:jpg,jpeg,png'
```
Comment (DungLe1603) [inline app/Http/Requests/UpdateProductRequest.php:33]: 300?

**Review by szabogaliakos — APPROVED** [review]

**Pull request merged**